### PR TITLE
Added library config

### PR DESCRIPTION
**Pull request by popenta**
Added a `LibraryConfig` that stores the default `hrp` value.

### CODE REVIEW

File: examples/Cookbook.ipynb [hunk 1]
```diff
@@ -184,6 +184,47 @@
     "print(\"Is contract:\", address.is_smart_contract())"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Changing the default hrp\n",
+    "\n",
+    "We have a configuration class, called `LibraryConfig`, that only stores the **default hrp** of the addresses. The default value is `erd`. The hrp can be changed when instantiating an address, or it can be changed in the `LibraryConfig` class, and all the addresses created will have the newly set hrp.  "
```
Comment (andreibancioiu) [inline examples/Cookbook.ipynb:193]: `that only stores ... ` - "for now" / "for the moment".

Maybe, in the future, we'll move the most of the constants (that we use for default values in various parts). That way, the constants will be "promoted" to static variables.

File: multiversx_sdk/core/address.py [hunk 2]
```diff
@@ -36,7 +37,7 @@ def __init__(self, pubkey: bytes, hrp: str) -> None:
             raise ErrBadPubkeyLength(len(pubkey), PUBKEY_LENGTH)
 
         self.pubkey = bytes(pubkey)
-        self.hrp = hrp
+        self.hrp = hrp if hrp else LibraryConfig.default_address_hrp
```
Comment (andreibancioiu) [inline multiversx_sdk/core/address.py:40]: All right, good change.

File: multiversx_sdk/core/config.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+@dataclass
+class LibraryConfig:
+    """
+    Global configuration of the library.
+
+    Generally speaking, this configuration should only be altered on exotic use cases
+    it can be seen as a collection of constants (or , to be more precise, rarely changed variables) that are used throughout the library.
+
+    Never alter the configuration within a library!
```
Comment (andreibancioiu) [inline multiversx_sdk/core/config.py:12]: Indeed. Only in an _end_ application.

Altering within a library (several times) might lead to ambiguity and conflicts.

File: multiversx_sdk/core/config.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    Only alter the configuration(if needed) within an(end) application that uses this library.
```
Comment (andreibancioiu) [inline multiversx_sdk/core/config.py:13]: Typo / spacing.

File: multiversx_sdk/core/constants.py [hunk 5]
```diff
@@ -11,9 +11,19 @@
 EGLD_TOKEN_IDENTIFIER = "EGLD"
 EGLD_NUM_DECIMALS = 18
 
+# left it for compatibility reasons, will be deleted in the future
 DELEGATION_MANAGER_SC_ADDRESS = "erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqylllslmq6y6"
+
+# left it for compatibility reasons, will be deleted in the future
```
Comment (andreibancioiu) [inline multiversx_sdk/core/constants.py:17]: :+1:

File: multiversx_sdk/network_providers/proxy_network_provider.py [hunk 6]
```diff
@@ -39,13 +38,12 @@ def __init__(
...
             auth: Union[AuthBase, None] = None,
-            address_hrp: str = DEFAULT_ADDRESS_HRP,
+            address_hrp: Optional[str] = None,
             config: Optional[NetworkProviderConfig] = None
     ) -> None:
         self.url = url
         self.auth = auth
-        self.address_hrp = address_hrp
-        self.address_hrp = address_hrp or DEFAULT_ADDRESS_HRP
+        self.address_hrp = address_hrp or LibraryConfig.default_address_hrp
```
Comment (andreibancioiu) [inline multiversx_sdk/network_providers/proxy_network_provider.py:46]: :+1:

File: multiversx_sdk/network_providers/transactions.py [hunk 7]
```diff
@@ -202,3 +202,41 @@ def to_dictionary(self) -> Dict[str, Any]:
             "gasPrice": self.gas_price,
             "data": self.data
         }
+
+
+# this is duplicated code, added here to get rid of a circular dependency
+# will be removed in V1
```
Comment (andreibancioiu) [inline multiversx_sdk/network_providers/transactions.py:208]: :+1:

**Review by andreibancioiu — DISMISSED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**